components/sysutils/virtualbox/patches/20-ctf.patch

```diff
@@ -0,0 +1,235 @@
+--- VirtualBox-6.1.22/kBuild/tools/GCC3PLAIN.kmk	1970-01-01 00:00:00
++++ VirtualBox-6.1.22/kBuild/tools/GCC3PLAIN.kmk	1970-01-01 00:00:00
+@@ -66,18 +66,37 @@ endif
+ # General Properties used by kBuild
+ TOOL_GCC3PLAIN_COBJSUFF         ?= .o
+ TOOL_GCC3PLAIN_CFLAGS           ?=
+-TOOL_GCC3PLAIN_CFLAGS.debug     ?= -g
++TOOL_GCC3PLAIN_CFLAGS.debug     ?= \
++	-gdwarf-2 -fno-inline-small-functions \
++	-fno-inline-functions-called-once \
++	-fno-ipa-cp -fno-ipa-icf \
++	-fno-clone-functions -fno-reorder-functions \
++	-fno-shrink-wrap -fno-aggressive-loop-optimizations
+ TOOL_GCC3PLAIN_CFLAGS.profile   ?= -O2 #-g -pg
+-TOOL_GCC3PLAIN_CFLAGS.release   ?= -O2
++TOOL_GCC3PLAIN_CFLAGS.release   ?= -O2 \
++	-gdwarf-2 -fno-inline-small-functions \
++	-fno-inline-functions-called-once \
++	-fno-ipa-cp -fno-ipa-icf \
++	-fno-clone-functions -fno-reorder-functions \
++	-fno-shrink-wrap -fno-aggressive-loop-optimizations
+ TOOL_GCC3PLAIN_CINCS            ?=
+ TOOL_GCC3PLAIN_CDEFS            ?=
+ 
+ TOOL_GCC3PLAIN_CXXOBJSUFF       ?= .o
+ TOOL_GCC3PLAIN_CXXOBJSUFF       ?= .o
+ TOOL_GCC3PLAIN_CXXFLAGS         ?=
+-TOOL_GCC3PLAIN_CXXFLAGS.debug   ?= -g
++TOOL_GCC3PLAIN_CXXFLAGS.debug   ?= \
++	-gdwarf-2 -fno-inline-small-functions \
++	-fno-inline-functions-called-once \
++	-fno-ipa-cp -fno-ipa-icf \
++	-fno-clone-functions -fno-reorder-functions
+ TOOL_GCC3PLAIN_CXXFLAGS.profile ?= -O2 #-g -pg
+-TOOL_GCC3PLAIN_CXXFLAGS.release ?= -O2
++TOOL_GCC3PLAIN_CXXFLAGS.release ?= -O2 \
++	-gdwarf-2 -fno-inline-small-functions \
++	-fno-inline-functions-called-once \
++	-fno-ipa-cp -fno-ipa-icf \
++	-fno-clone-functions -fno-reorder-functions \
++	-fno-shrink-wrap -fno-aggressive-loop-optimizations
+ TOOL_GCC3PLAIN_CXXINCS          ?=
+ TOOL_GCC3PLAIN_CXXDEFS          ?=
+ 
+@@ -91,6 +110,10 @@ TOOL_GCC3PLAIN_ARLIBSUFF        ?= .a
+ 
+ TOOL_GCC3PLAIN_LDFLAGS          ?=
+ 
++TOOL_CTFCONVERT			?= /opt/onbld/bin/i386/ctfconvert
++TOOL_CTFMERGE			?= /opt/onbld/bin/i386/ctfmerge
++TOOL_STRIP			?= /usr/bin/strip
++
+ 
+ ## Compile C source.
+ # @param    $(target)   Normalized main target name.
+@@ -122,6 +145,9 @@ define TOOL_GCC3PLAIN_COMPILE_C_CMDS
+ 		$(flags) -fpreprocessed -x c\
+ 		-o $(obj)\
+ 		-
++if ($(OOCEVER) >= 151034)
++	$(QUIET)$(TOOL_CTFCONVERT) -mki -l vbox $(obj)
```

Comment:
These kind of ifs is what Alexander mentioned. We don't have OOCEVER defined. This is a guard to guarantee that the OmniOS's ooce tools (and the corresponding OmniOS version) is recent enough.
As we have nightly illumos-gate builds it would probably sufficient to remove these guards and keep the ctfconvert calls IF we want to do that (another thing Alexander has asked).